File: cplusplus/first100primes.cc
#include <iostream>
#include <cmath>

using namespace std;

bool is_prime(int n)
{
    if (n % 2 == 0) {
        return false;
    } else {
        for (int y = 3; y <= ceil(sqrt(n)); y += 2) {
            if (n % y == 0) {
                return false;
            }
        }
        return true;
    }
}

int main()
{
    for (int found = 0, candidate = 3; found < 100; candidate++) {
        if (is_prime(candidate)) {
            cout << candidate << " ";
            found++;
        }
    }
    cout << flush << endl;
}

File: python/first100primes.py
#!/usr/bin/env python

import math

N_PRIMES = 100

def next_prime():
    n = 3
    while True:
        while not is_prime(n):
            n += 2
        else:
            yield n
            n += 2

def is_prime(x):
    for y in xrange(2, int(math.ceil(math.sqrt(x)) + 1)):
        if x % y == 0:
            return False
    else:
        return True

np = next_prime()
print [np.next() for p in range(N_PRIMES)]
